docs/lib/datastore/fragments/android/data-access.md

```diff
@@ -9,75 +16,71 @@ Post post = Post.builder()
     .status(PostStatus.ACTIVE)
     .build();
 
-Amplify.DataStore.save(post, new ResultListener<DataStoreItemChange<Post>>() {
-    @Override
-    public void onResult(DataStoreItemChange<Post> result) {
-        Log.i("DataStore", "Result: " + result);
-    }
-
-    @Override
-    public void onError(Throwable error) {
-        Log.e("DataStore", "Error.", error);
-    }
-});
+Amplify.DataStore.save(post,
+    saved -> Log.i("DataStore", "Saved a post: " + saved.item()),
+    saveFailure -> Log.e("DataStore", "Save failed.", saveFailure)
+);
 ```
 
 ## Query Data
 
-Queries are always performed against the local copy of your data. When connected to the cloud, the local copy of your data is automatically updated in the background. You can query for models by their class, or by providing additional search criteria for finer-grained results. A simple query without any filtering criteria is shown below.
+Queries are always performed against the local copy of your data. When
+connected to the cloud, the local copy of your data is automatically
+updated in the background. You can query for models by their class, or
+by providing additional search criteria for finer-grained results. A
+simple query without any filtering criteria is shown below.
 
 ```java
-Amplify.DataStore.query(Post.class, new ResultListener<Iterator<Post>>() {
-    @Override
-    public void onResult(Iterator<Post> result) {       
-        while (result.hasNext()) {
-            Post post = result.next();
+Amplify.DataStore.query(Post.class,
+    allPosts -> {
+        while (allPosts.hasNext()) {
+            Post post = allPosts.next();
             Log.i("DataStore", "Title: " + post.getTitle());
             Log.i("DataStore", "Rating: " + post.getRating());
             Log.i("DataStore", "Status: " + post.getStatus());
         }
-    }
-
-    @Override
-    public void onError(Throwable error) {
-        Log.e("DataStore", "Error.", error);
-    }
-});
+    },
+    queryFailure -> Log.e("DataStore", "Query failed.", queryFailure)
+); 
 ```
 
 ### Query with Predicates
 
-You can provide additional filters to your query using a **query predicate**. The AWS-standard nomenclature of these query predicates will be familiar to those who have used Amazon DynamoDB, in the past.
+You can provide additional filters to your query using a **query
+predicate**. The AWS-standard nomenclature of these query predicates
```

Comment:
I know @drochetti is also thinking about DataStore docs across the platforms, so this might be worth flagging for our next pass rather than changing it now. "AWS-standard nomenclature" is an overstatement. These are operators used in conditions passed to query or scan operations against DynamoDB.